Assignment4/task07_alvarohdezrodri.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/FacultadInformatica-LinkedData/Curso2022-2023-DataScience/blob/master/Assignment4/course_materials/notebooks/Task07.ipynb
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+#RDFLib
+print("RDFLib")
+ns = Namespace("http://somewhere#")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s, "subclase de Person.")
+  for s2, p2, o2 in g.triples((None, RDFS.subClassOf, s)):
+    print(s2, "es subclase de", s, "(subclase de Person).")
+print()
+
+#SPARQL
+print("SPARQL")
+q1 = prepareQuery('''
+SELECT DISTINCT ?subclass
+WHERE{
+?subclass rdfs:subClassOf* <http://somewhere#Person>.
```

Comment:
the predicate * is for 0 or more repetitions. You want to use + here, which is 1 or more